File: app/src/main/java/com/bopr/android/smailer/AppBackupAgent.java
package com.bopr.android.smailer;

import android.app.backup.BackupAgentHelper;
import android.app.backup.SharedPreferencesBackupHelper;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Backup agent. Required on Android 2.2 (API level 8) to Android 6.0 (API level 23).
 *
 * @author Boris Pronin (<a href="mailto:boprsoft.dev@gmail.com">boprsoft.dev@gmail.com</a>)
 */
public class AppBackupAgent extends BackupAgentHelper {

    private static Logger log = LoggerFactory.getLogger("Backup");

    @Override
    public void onCreate() {
        log.debug("Create");
        addHelper("preferences", new SharedPreferencesBackupHelper(this, Settings.PREFERENCES_STORAGE_NAME));
//        addHelper("keystore", new FileBackupHelper(this, "../databases/" + Cryptor.KEYSTORE_FILE));
//        addHelper("database", new FileBackupHelper(this, "../databases/" + Settings.DB_NAME));
    }

}

File: app/src/main/kotlin/com/bopr/android/smailer/AppBackupAgent.kt
package com.bopr.android.smailer

import android.app.backup.BackupAgentHelper
import android.app.backup.SharedPreferencesBackupHelper
import com.bopr.android.smailer.Settings.Companion.sharedPreferencesName
import com.bopr.android.smailer.util.Logger

/**
 * Backup agent. Required on Android 2.2 (API level 8) to Android 6.0 (API level 23).
 *
 * @author Boris Pronin ([boprsoft.dev@gmail.com](mailto:boprsoft.dev@gmail.com))
 */
@Suppress("unused")
class AppBackupAgent : BackupAgentHelper() {

    override fun onCreate() {
        log.debug("Created")

        addHelper("settings", SharedPreferencesBackupHelper(this, sharedPreferencesName))
        //        addHelper("database", new FileBackupHelper(this, "../databases/" + Settings.DB_NAME));
    }

    companion object{
        private val log = Logger("Backup")
    }
}